src/test/java/org/jenkinsci/plugins/docker/workflow/DockerDSLTest.java

```diff
@@ -325,6 +325,38 @@ private static void grep(File dir, String text, String prefix, Set<String> match
         });
     }
 
+    @Test public void buildWithMultiStage() {
```

Comment:
If this is going to break with Docker < 17.05, could we put in an `assume` like https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/AbstractDeclarativeTest.java#L218-L227? With the body of that method copied in and the version numbers switched, you should be able to ensure this test only gets run when on Docker 17.05 or later.